src/test/java/model/CarsTest.java

```diff
@@ -0,0 +1,60 @@
+package model;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class CarsTest {
+    private static final int INITIAL_POSITION = 0;
+
+    private List<Car> setUpTestCars(String input) { // 테스트를 위한 Cars 설정
+        return Arrays.stream(input.split(","))
+            .map(Car::new)
+            .collect(Collectors.toList());
+    }
+
+    @DisplayName("Cars 객체 불변 테스트")
+    @ParameterizedTest
+    @ValueSource(strings = {"als,qweop,skld", "car1,car2,car3"})
+    void newCarsTest(String input) {
+        Cars cars = new Cars(setUpTestCars(input));
+        assertThatExceptionOfType(UnsupportedOperationException.class)
+            .as("Cars 객체는 변경할 수 없습니다.")
+            .isThrownBy(() -> cars.getCars().remove(0));
+    }
+
+    @DisplayName("경주 진행중 Car객체가 갖는 position값 유효성 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"car1,car2:5", "carr1,carr2:3"}, delimiter = ':')
+    void raceTest(String input, String inputMoveTimes) {
+        List<Car> testCars = setUpTestCars(input);
+        Cars cars = new Cars(testCars);
+        int moveTimes = Integer.valueOf(inputMoveTimes);
+        for (int i = 0; i < moveTimes; i++) {
+            cars.race();
+        }
+        assertThat(cars.getCars())
+            .extracting(Car::getPosition)
+            .allMatch(position -> validatePosition(position, moveTimes));
+
+    }
+
+    private boolean validatePosition(int position, int moveTimes) {
+        return INITIAL_POSITION <= position && position <= moveTimes;
+    }
+
+    @DisplayName("자동차 게임 우승자 이름 얻기 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"car1,car2,car3:car1, car2, car3"}, delimiter = ':')
+    void getWinnersTest(String input, String winners) {
+        Cars cars = new Cars(setUpTestCars(input));
+        assertThat(cars.getWinners()).isEqualTo(winners);
```

Comment:
깊은 복사와 얕은 복사 얘기를 하기에 앞서서 이 테스트가 의도한대로 동작을 하는 코드인지 잘 모르겠습니다. 
``자동차 게임 우승자 이름을 얻는 테스트``라고 했는데, Cars 객체의 **getWinners()** 메소드가 실질적으로 그냥 Cars의 상태값을 그대로 보여주기 때문입니다(car1, car2, car3는 모두 position이 0으로 동일). Cars의 상태값으로 가진 car 객체들의 position값이 다른 경우를 테스트해야할 것 같습니다. (같은 경우도 있다고해서 나쁠건 없는 것 같습니다)  

```suggestion
        Cars cars = new Cars(setUpTestCars(input));
        assertThat(cars.getWinners())
                .usingFieldByFieldElementComparator()
                .containsExactly(new Car("car1"), new Car("car2"), new Car("car3"));
```

식으로도 변경이 가능할 것 같네요. (**CsvSource** 를 꼭 이용해야 할 필요는 없다고 생각합니다)  List<객체>와 List<객체> 타입의 비교도 이런식으로 가능하다는 점 정도를 캐치하시면 좋을 것 같네요.

Reply:
좋은 예시 감사합니다. 많이 배운 것 같습니다.